test/location.test.js

```diff
@@ -104,6 +145,28 @@ define(function(require) {
 
           location.on('statechange.test-no-dispatch', null)
         })
+
+        it('should preserve query parameters', function() {
+          var loc = window.location
+          var originalQuery = new URLSearchParams(loc.search.substring(1))
+
+          originalQuery.set('foo', 'bar')
+          originalQuery.set('fizz', 'buzz')
+
+          window.history.replaceState(
+            null, null,
+            loc.origin + loc.pathname + '?' + originalQuery.toString()
+          )
+
+          location.basePath('/base')
+          location.getState()
+          location.pushState('/base/foo')
+
+          var newQuery = new URLSearchParams(window.location.search.substring(1))
+
+          expect(newQuery.get('foo')).to.equal('bar')
+          expect(newQuery.get('fizz')).to.equal('buzz')
```

Comment:
Same here, couldn't we just `expect(location.getState()).to.equal('/base/foo?foo=bar&fizz=buzz')`? By the way, should `location.pushState` really preserve the query parameters like this? I'd expect any query parameters to be lost once a new state is pushed, unless those are indeed specified in the new state.